lib/Runtime/Language/JavascriptOperators.cpp

```diff
@@ -2307,6 +2307,11 @@ namespace Js
 
     BOOL JavascriptOperators::SetProperty(Var receiver, RecyclableObject* object, PropertyId propertyId, Var newValue, PropertyValueInfo * info, ScriptContext* requestContext, PropertyOperationFlags propertyOperationFlags)
     {
+        if (propertyId == PropertyIds::next && object == requestContext->GetLibrary()->GetArrayIteratorPrototype())
+        {
+            requestContext->GetLibrary()->SetArrayIteratorHasUserDefinedNext(true);
+        }
+
```

Comment:
Makes sense. Returning to using GetProperty() and comparison of `%ArrayIteratorPrototype%.next` to its built-in function. Calling GetProperty() with implicit call disabled. Updated in latest rev.